sdks/go/pkg/beam/core/runtime/metricsx/metricsx.go

```diff
@@ -150,7 +183,7 @@ func extractGaugeValue(reader *bytes.Reader) (metrics.GaugeValue, error) {
 }
 
 func newLabels(miLabels map[string]string) *metrics.Labels {
-	labels := metrics.UserLabels(miLabels["PTRANSFORM"], miLabels["NAMESPACE"], miLabels["NAME"])
+	labels := metrics.UserLabels(miLabels["PTRANSFORM"], miLabels["NAMESPACE"], miLabels["NAME"], miLabels["PCOLLECTION"])
```

Comment:
Related to the comment about modifying UserLabels, consider using wether PCollection is populated to determine if `metrics.PCollectionLabels` is used instead.